timelock-impl/src/test/java/com/palantir/atlasdb/timelock/paxos/PaxosTimestampBoundStoreTest.java

```diff
@@ -315,61 +249,161 @@ public void canGetAgreedStateAfterNodeDown() {
         assertThat(store.forceAgreedState(1, 1L)).isEqualTo(ONE_AND_FORTY_TWO);
         assertThat(store.getAgreedState(1)).isEqualTo(ONE_AND_FORTY_TWO);
     }
 
-    @Test
-    public void forceAgreedStateOperatesAtSequenceNumberLevel() {
+    @ParameterizedTest
+    @MethodSource("data")
+    public void forceAgreedStateOperatesAtSequenceNumberLevel(boolean useBatch) {
+        setup(useBatch);
         long fortyThree = FORTY_TWO + 1;
         assertThat(store.forceAgreedState(1, FORTY_TWO)).isEqualTo(ONE_AND_FORTY_TWO);
         assertThat(store.forceAgreedState(0, fortyThree)).isEqualTo(ImmutableSequenceAndBound.of(0, fortyThree));
         assertThat(store.getAgreedState(0)).isEqualTo(ImmutableSequenceAndBound.of(0, fortyThree));
         assertThat(store.getAgreedState(1)).isEqualTo(ONE_AND_FORTY_TWO);
     }
 
-    @Test
-    public void forceAgreedStateThrowsIfNoStateWasAgreedUpon() {
+    @ParameterizedTest
+    @MethodSource("data")
+    public void forceAgreedStateThrowsIfNoStateWasAgreedUpon(boolean useBatch) {
+        setup(useBatch);
         assertThatThrownBy(() -> store.forceAgreedState(1, null)).isInstanceOf(NullPointerException.class);
     }
 
-    @Test
-    public void retriesForceAgreedStateUntilSuccessful() throws Exception {
+    @ParameterizedTest
+    @MethodSource("data")
+    public void retriesForceAgreedStateUntilSuccessful(boolean useBatch) throws Exception {
+        setup(useBatch);
         PaxosProposer wrapper = spy(new OnceFailingPaxosProposer(createPaxosProposer(0)));
         store = createPaxosTimestampBoundStore(0, wrapper);
         store.forceAgreedState(1, TIMESTAMP_1);
         assertThat(store.getUpperLimit()).isGreaterThanOrEqualTo(TIMESTAMP_1);
         verify(wrapper, times(2)).propose(anyLong(), any(byte[].class));
     }
 
+    public void setup(boolean useBatch) {
+        List<PaxosAcceptor> acceptors = new ArrayList<>();
+        List<BatchPaxosAcceptor> batchPaxosAcceptors = new ArrayList<>();
+        List<BatchPaxosLearner> batchPaxosLearners = new ArrayList<>();
+
+        for (int i = 0; i < NUM_NODES; i++) {
+            String root = temporaryFolder.getAbsolutePath();
+            LocalPaxosComponents components = LocalPaxosComponents.createWithAsyncMigration(
+                    TimelockPaxosMetrics.of(PaxosUseCase.TIMESTAMP, MetricsManagers.createForTests()),
+                    PaxosUseCase.TIMESTAMP,
+                    Paths.get(root, i + "legacy"),
+                    SqliteConnections.getDefaultConfiguredPooledDataSource(Paths.get(root, i + "sqlite")),
+                    UUID.randomUUID(),
+                    true,
+                    OrderableSlsVersion.valueOf("0.0.0"),
+                    false);
+
+            AtomicBoolean failureController = new AtomicBoolean(false);
+            failureToggles.add(failureController);
+
+            learners.add(ToggleableExceptionProxy.newProxyInstance(
```

Comment:
It is unfortunate that we cannot parametrise `@BeforeEach` out-of-the-box in Junit5 (No vanilla support). There are some custom solutions I found but they don't worth the pain.